### ISSUE TITLE
yolov3.weights cannot be found and training format

### ISSUE THREAD

**Issue by CodLiver** [andy-yun/pytorch-0.4-yolov3#43]
I was planning to train my own data, but it says yolov3.weights not found. I checked the master repo of you and I didnt find that too. Was it accidentally removed or not meant to happen?

I had the weights earlier, but wondered what happened.

Besides, What is the format of training our custom data? 

I have 

train/class0

train/class1
 
and 

val/

that has the image#.jpg

edit: I love this code and thanks for everything!

**Comment by andy-yun** [comment]
You can find the way to get yolo3.weights at https://pjreddie.com/darknet/yolo/  and you can find the label information of VOC.
The yolo site can help you.

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by CodLiver** [comment]
Thanks! I have checked it and tried to do bounding box on my dataset.

When it said: "relative to image", does it mean range between (0,1) or in terms of px? I used your VOC formatter to have ranges between (0,1), but I still get various errors unfortunately. I think those errors happen because I changed some parts of the code to integrate my dataset. 

according to VOC format: `class,x,y,width,height`.

However, then dataset.py throws error by saying that it couldnt find the image. Hence I added `class,x,y,width,height,**imagepath**`

Anyway..

Besides, when I was checking the code and formatting to my own dataset.(probably errors' reason), I come across a different type of error that might be helpful:

I think 

        loss_coord = nn.MSELoss(reduction='sum')(coord*coord_mask, tcoord*coord_mask)/nB

        loss_conf  = nn.BCELoss(reduction='sum')(conf*conf_mask, tconf*conf_mask)/nB

        loss_cls   = nn.BCEWithLogitsLoss(reduction='sum')(cls, tcls)/nB

should be changed to

        loss_coord = nn.MSELoss(**reduce=True**)(coord*coord_mask, tcoord*coord_mask)/nB

        loss_conf  = nn.BCELoss(**reduce=True**)(conf*conf_mask, tconf*conf_mask)/nB

        loss_cls   = nn.BCEWithLogitsLoss(**reduce=True**)(cls, tcls)/nB

        # without **...**


Because in my "nn" library, it has 2 optional parameter which "reduction" is changed to "reduce (bool)". I planned to do a pull request, but I wasn't sure if that is applicable to you as well. 

Hope it helps! :)

**Comment by andy-yun** [comment]
At first, image bounding box is represented as the relative value to original image and is placed between [0,1].

Second, **reduction='sum'** is identical to **size_average=False** in pytorch 0.4 or below version.
**reduction='sum'** is introduced at pytorch 0.4.1 or over (the latest stable version is pytorch 1.0)

Third, I recommend you should download again the source code. The yolo_layer.py is modified to improve performance. Some loss functions are changed.

Bye.

**Comment by CodLiver** [comment]
oh thanks for the elaborate explanation. I should update my pytorch then.

I am sorry, my bad.

Best regards,
